File: stb_grpc_demo/user-service/src/main/java/com/stb/user/repository/UserRepository.java
package com.stb.user.repository;

import com.stb.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface UserRepository extends JpaRepository<User, String> {
    List<User> findAllByNameAndAccountNumberAndBankName(String name, String accountNumber, String bankName);
}

File: h2database/src/main/kotlin/com/h2database/repository/UserRepository.kt
package com.h2database.repository

import com.h2database.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<UserEntity?, Long?> {
    fun findByUsername(username: String?): UserEntity?
}